app/server/app/routes/data.js

```diff
@@ -23,6 +23,14 @@ class InvalidParameterException extends Error {
   }
 }
 
+function appendLatestToWhere(query, column, columnType, baseQuery) {
+  if (columnType !== "numeric" && columnType !== "timestamptz") return;
+
+  const subQuery = baseQuery.clone();
+  query.where(column, subQuery.max(column));
```

Comment:
```suggestion
  const subQuery = baseQuery.clone();
  subQuery.select("organizationid").max(column).groupBy("organizationid");
  query.whereIn(["organizationid", column], subQuery);
```

Nothing to change here. We will probably want to bring this up in a future meeting. I'm thinking EPA might want something a litte bit more complicated than this. 

This query just gets the data that has the latest reporting cycle in the system, which is implicitly filtering out organizations that don't have data in the most recent reporting cycle. If you download latest from the sources profile you won't get any data from West Virginia, because it's latest reporting cycle is 2016 instead of 2022. 

The above suggested code will get the latest data for every organization, which I think is what EPA wants. This basically makes a query that looks like the following:

```
SELECT * 
FROM assessments
WHERE (organizationid, reportingcycle) IN (
  SELECT organizationid, MAX(reportingcycle)
  FROM assessments
  GROUP BY organizationid
)
``` 

Below, are the row counts each table will give if a user goes in and clicks download without any filters. Based on the size of these we may need to pre-build these exports. I estimate it would take about 3 - 5 hours to generate csv downloads for all of these which would be posted on our national downloads page.
* actions - 3,480,216 (does not have reporting cycle column)
* assessment_units - 1,316,518
* assessment_units_monitoring_locations - 1,645,620
* assessments - 54,166,918 (it is 54,163,082 if we filter out all of the retired and historacle items)
* catchment_correspondence - 65,139,462
* sources - 2,663,302
* tmdl - 73,603

Maybe EPA can give us another field in the MVs that we can use to filter this down further.

Reply:
Hmm good point, I was thinking we just wanted one cycle across the board. I guess we'll wait and ask.